airflow/jobs/base_job.py

```diff
@@ -246,7 +246,6 @@ def run(self):
             finally:
                 self.end_date = timezone.utcnow()
                 session.merge(self)
-                session.commit()
```

Comment:
this one is safe to remove since its the last line within a create_session context manager, so the commit is happening right after this anyways.

Reply:
Is it? The context manager only commits when exited successfully (rollbacks on an exception). This commit also saves changes on an exception.

Reply:
Good catch, I'll add this one back in.